src/main/java/com/main/server/service/impl/ProductServiceImpl.java

```diff
@@ -43,20 +43,20 @@ public Product saveProduct(Product product, Long orgId) throws ResourceAlreadyEx
     @Override
     public Product getProductById(Long id) {
         log.info("Enter getProductById() productId: {}", id);
-        return productRepository.getById(id).orElseThrow(
+        return productRepository.findById(id).orElseThrow(
                 () -> new ResourceNotFoundException(
                         String.format("Product with such id {%d} is not exist", id))
         );
     }
 
     @Override
-    public Product updateProduct(Product product, Long orgId) {
+    public Product updateProduct(Product product, Long orgId, Long id) {
         log.info("Enter updateProduct() productName: {}", product.name());
-        Product exitedProduct = getProduct(product.getId());
+        Product exitedProduct = getProduct(id);
 
         Product exitedProductWithSameName = productRepository.findByNameAndOrganizationId(product.name(), orgId)
                 .orElse(null);
-        if (exitedProductWithSameName != null) {
+        if (exitedProduct.equals(exitedProductWithSameName) && !isDeleted(product)) {

```

Comment:
I think better to get only not deleted products from db than use specific method to check deletedBy